torch/csrc/jit/codegen/cuda/lower_fusion_simplifier.cpp

```diff
@@ -26,51 +27,56 @@ class UnaryOpInserter : private kir::ExprMutator {
     kir::ExprMutator::traverseAndInsert(exprs);
   }
 
+  void registerReplaceAndPropagate(Expr* old_expr, Expr* new_expr) {
```

Comment:
I'm thinking of a different design of keeping the information from becoming stale. One quick way would be associate the information with `TensorView` rather than `Expr` as we don't replace `TensorView` while lowering. Will have a follow-up PR.